{{cookiecutter.project_slug}}/src/{{cookiecutter.pkg_name}}/__init__.py

```diff
@@ -5,9 +5,32 @@
 
 from importlib.metadata import PackageNotFoundError, version
 
+# try:
+#     __version__ = version('{{ cookiecutter.project_slug }}')
+# except PackageNotFoundError:  # pragma: no cover
+#     __version__ = 'unknown'
+# finally:
+#     del version, PackageNotFoundError
+
+# up-to-date version tag for modules installed in editable mode inspired by
+# https://github.com/maresb/hatch-vcs-footgun-example/blob/main/hatch_vcs_footgun_example/__init__.py
+# Define the variable '__version__':
 try:
-    __version__ = version('{{ cookiecutter.project_slug }}')
-except PackageNotFoundError:  # pragma: no cover
-    __version__ = 'unknown'
-finally:
-    del version, PackageNotFoundError
+
+    # own developed alternative variant to hatch-vcs-footgun overcoming problem of ignored setuptools_scm settings
+    # from hatch-based pyproject.toml libraries
+    from hatch.cli import hatch
+    from click.testing import CliRunner
```

Comment:
This is required for retrieving the actual build version. It is an adjusted verison of the footgun as commented.

Reply:
Not sure I really get it. My original code accomplished the same thing. It read out the version but only using modules that are shipped directly with Python. So the big advantage of my original code is that you don't need to have hatch installed to use the package. I know that developers of a package need to have hatch installed but you should always think about the users of what you are building. They should not be required to install hatch.

Reply:
If you do an "pip install -e" then you always see the last install version -- not the last commit version.
It is explained here:
https://github.com/maresb/hatch-vcs-footgun-example

But I agree: this is probably a small advantage compared to the setuptool_scm dependency. And it is only neceesary if one has very strict version dependencies in the code.
Probably refering to metadata or the local _version.py after a hatch build is a good compomise?

Reply:
Argh, I did miss that I eliminated the setuptools_scm dependency already because of using hatch compared to the original footgun-version. the setuptools_scm did not work correctly with the version bumping.